e2e/cypress/integration/messaging/channel_and_posts_links_spec.js

```diff
@@ -0,0 +1,232 @@
+        cy.getNthPostId(-index).then((postId) => {
+            // # Scroll into view
+            cy.get(`#post_${postId}`).scrollIntoView();
+        });
+
+        // # Search for a message in the current channel
+        cy.get('#searchBox').clear().type('in:town-square').type('{enter}');
+
+        // # Jump to first permalink view (most recent message)
+        cy.get('.search-item__jump').first().click();
+
+        // # Verify that we jumped to the last message
+        cy.getLastPostId().then((postId) => {
+            cy.get(`#postMessageText_${postId}`).should('contain', index - 1);
+        });
+    });
+
+    it('MM-T2222 - Channel shortlinking - ~ autocomplete', () => {
+        const publicChannelName = 'town-square';
+        const publicChannelDisplayName = 'Town Square';
+
+        cy.visit(`/${testTeam.name}/channels/off-topic`).wait(TIMEOUTS.FIVE_SEC);
+
+        // # Clear then type ~ and prefix of channel name
+        cy.get('#post_textbox').should('be.visible').clear().type('~' + publicChannelName.substring(0, 3));
+
+        // * Verify that the item is displayed or not as expected.
+        cy.get('#suggestionList').should('be.visible').within(() => {
+            cy.findByText(publicChannelDisplayName).should('be.visible');
+        });
+
+        // # Post channel mention
+        cy.get('#post_textbox').type('{enter}{enter}');
+
+        // # Check that the user name has been posted
+        cy.getLastPostId().then((postId) => {
+            cy.get(`#postMessageText_${postId}`).should('contain', publicChannelDisplayName);
+            cy.get('a.mention-link').click();
+            cy.get('#channelHeaderTitle').should('be.visible').should('contain', publicChannelDisplayName);
+        });
+    });
+
+    it('MM-T2224 - Channel shortlinking - link joins public channel', () => {
+        cy.visit(`/${testTeam.name}/channels/town-square`).wait(TIMEOUTS.FIVE_SEC);
+
+        // # Clear then type ~ and prefix of channel name
+        cy.get('#post_textbox').should('be.visible').clear().type(`~${testChannel.display_name}`);
+
+        // * Verify that the item is displayed or not as expected.
+        cy.get('#suggestionList').within(() => {
+            cy.findByText(`${testChannel.display_name}`).should('be.visible');
+        });
+
+        // # Post channel mention
+        cy.get('#post_textbox').type('{enter}{enter}');
+
+        cy.apiLogout();
+        cy.apiLogin(notinchannelUser);
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        // # Check that the user display name has been posted
+        cy.getLastPostId().then((postId) => {
+            cy.get(`#postMessageText_${postId}`).should('contain', `${testChannel.display_name}`);
+            cy.get('a.mention-link').click();
+            cy.get('#channelHeaderTitle').should('be.visible').should('contain', `${testChannel.display_name}`);
+        }).then(() => {
+            cy.apiLogout();
+            cy.apiLogin(sysadmin);
```

Comment:
There's a command `cy.apiAdminLogin();`. Also, can we put this in the `beforeEach`? Reason being is that this is not guaranteed to run always, in case something failed in the middle.